File: stack.cpp
#include <iostream>
#include <stack>

using namespace std;

int main(){

	stack<int> s;


	// top
	cout << "top element: " << s.top() << '\n';

  // push
	s.push(1);
	s.push(2);
	s.push(3);

	// pop
	s.pop(); // 3이 삭제


	// size
	cout << "스택 크기 " << s.size() << '\n';


	// empty
	cout << "비어있는지 확인 " << (s.empty() ? "Yes" : "No") << '\n';

	return 0;

}

File: stack.java
//제네릭 스택

class Gstack<T>{
  int top;
  Object [] stack;
  public Gstack(){
    top=0;
    stack=new Object(10);
  }
  public void push(T item){
    if(top==10)
      return;
    stack[top]=item;
    top++;
  }
  public T pop(){
    if(top==0){
      return null;
    top--;
    }
  }
  
public class Stack{
  
 public static void main(String[]args){
   
   Gstack<String> stringStack = new Gstack<String>();
   stringStack.push("Java");
   stringStack.psuh("stack");
   
   System.out.print(stringStack.pop());
   
   
  }
}
